Source/Orts.Simulation/Simulation/RollingStocks/SubSystems/CruiseControl.cs

```diff
@@ -0,0 +1,1458 @@
+                if (Locomotive.DynamicBrakePercent > 100)
+                    Locomotive.DynamicBrakePercent = 100;
+                ForceThrottleAndDynamicBrake = Locomotive.DynamicBrakePercent;
+
+            UpdateMotiveForce(elapsedClockSeconds, AbsWheelSpeedMpS);
+        }
+
+        public void Save(BinaryWriter outf)
+        {
+            outf.Write(this.AntiWheelSpinEquipped);
+            outf.Write(this.applyingPneumaticBrake);
+            outf.Write(this.Battery);
+            outf.Write(this.brakeIncreasing);
+            outf.Write(this.clockTime);
+            outf.Write(this.controllerTime);
+            outf.Write(this.CurrentSelectedSpeedMpS);
+            outf.Write(this.currentThrottlePercent);
+            outf.Write(this.DeltaCoefficient);
+            outf.Write(this.DynamicBrakeDescentCoefficient);
+            outf.Write(this.DynamicBrakeMaxForceAtSelectorStep);
+            outf.Write(this.dynamicBrakeSetToZero);
+            outf.Write(this.fromAcceleration);
+            outf.Write(this.MaxAccelerationMpSS);
+            outf.Write(this.MaxDecelerationMpSS);
+            outf.Write(this.maxForceDecreasing);
+            outf.Write(this.maxForceIncreasing);
+            outf.Write(this.maxForceN);
+            outf.Write(this.nextSelectedSpeedMps);
+            outf.Write(this.restrictedRegionTravelledDistance);
+            outf.Write(this.RestrictedSpeedActive);
+            outf.Write(this.SelectedMaxAccelerationPercent);
+            outf.Write(this.SelectedMaxAccelerationStep);
+            outf.Write(this.SelectedMaxAccelerationStep);
+            outf.Write(this.SelectedNumberOfAxles);
+            outf.Write(this.SelectedSpeedMpS);
+            outf.Write((int)this.SpeedRegMode);
+            outf.Write(this.SpeedRegulatorMaxForcePercentUnits);
+            outf.Write(this.SpeedRegulatorMaxForceSteps);
+            outf.Write(this.SpeedRegulatorNominalSpeedStepMpS);
+            outf.Write((int)this.SpeedSelMode);
+            outf.Write(this.StartReducingSpeedDelta);
+            outf.Write(this.ThrottleDecreaseSpeed);
+            outf.Write(this.ThrottleIncreaseSpeed);
+            outf.Write(this.throttleIsZero);
+            outf.Write(this.trainBrakePercent);
+            outf.Write(this.trainLength);
+            outf.Write(this.UseThrottle);
+            outf.Write(this._AccelerationMpSS);
+            outf.Write(this.TrainLengthMeters);
+        }
+
+        public void Restore(BinaryReader inf)
+        {
+            AntiWheelSpinEquipped = inf.ReadBoolean();
+            applyingPneumaticBrake = inf.ReadBoolean();
+            Battery = inf.ReadBoolean();
+            brakeIncreasing = inf.ReadBoolean();
+            clockTime = inf.ReadDouble();
+            controllerTime = inf.ReadSingle();
+            CurrentSelectedSpeedMpS = inf.ReadSingle();
+            currentThrottlePercent = inf.ReadSingle();
+            float deltaCoefficient = inf.ReadSingle();
```

Comment:
What's the purpose for saving this parameters but then discard them?

Reply:
Will review.

Reply:
Usually the only things that should be saved are those that change during the simulation (i.e. selected cruise speed, cruise speed operation mode, etc.)

Reply:
Fixed in new PR

Reply:
You don't need to create more PRs. When you push the changes to your upstream repository, this PR is automatically updated.